src/main/java/ui/GUIController.java

```diff
@@ -102,11 +116,30 @@ public void panelFilterExpressionChanged(FilterPanel changedPanel) {
                 .collect(Collectors.toList());
     }
 
+    private void updateRemainingRateEvent(UpdateRateLimitsEvent e) {
+        updateAPIBox(e);
+    }
+
+    private void updateSyncRefreshRate(UpdateRateLimitsEvent e) {
+        lastNumberOfApiCallsUsed = computePreviousRemainingApiRequests(e);
+        refreshTimeInMins = ui.timerManager.computeTickerTimerPeriod(e.remainingRequests,
+                Utility.minutesFromNow(e.nextRefreshInMillisecs), lastNumberOfApiCallsUsed);
+        ui.timerManager.changeTickingTimerPeriodInMins((int) refreshTimeInMins);
+    }
+
+    private int computePreviousRemainingApiRequests(UpdateRateLimitsEvent e) {
```

Comment:
Passing the `Event` object to methods other than the direct event handler itself might make the method unclear from its signature. I think we can just pass what it needs here i.e. `e.remainingRequests` as something like `currentRemainingRequests`.

Also this method could have a comment explaining its intention.